src/bin/pg_autoctl/pgsql.c

```diff
@@ -1223,15 +1227,15 @@ pgsql_get_sync_state_and_wal_lag(PGSQL *pgsql, const char *slotName,
 	}
 
 	strlcpy(pgsrSyncState, context.syncState, PGSR_SYNC_STATE_MAXLENGTH);
-	*walLag = context.walLag;
+	strlcpy(currentLSN, context.currentLSN, bufferLength-1);
```

Comment:
Do we actually need the `-1` here? We certainly don't do that elsewhere, so that reads strange. Are you telling me we need to fix all the other use cases for `strlcpy` in the code?

Reply:
removed, it won't actually matter though.